### PR TITLE
Bug 1614919 browser.identity.launchWebAuthFlow() exposes redirect_url

### PR DESCRIPTION
**Pull request by rebloor**
#### Summary
Added notes to:
* getRedirectURL regarding the use of a loopback address
* launchWebAuthFlow  noting the need for redirect_uri to be set to the URL returned by identity.getRedirectURL()

#### Test results and supporting details

Ran npm test, no issues found

#### Related issues
[Bug 1614919](https://bugzilla.mozilla.org/show_bug.cgi?id=1614919)

### CODE REVIEW

File: webextensions/api/identity.json [hunk 1]
```diff
@@ -12,9 +12,15 @@
...
-              "firefox": {
-                "version_added": "53"
-              },
+              "firefox": [
+                {
+                  "version_added": "53"
+                },
+                {
+                  "version_added": "86",
+                  "notes": "The loopback address http://127.0.0.1/mozoauth2/[subdomain of URL returned by identity.getRedirectURL()] can be used as the redirect URL to enable the use of Google OAuth flows."
```
Comment (Rob--W) [inline webextensions/api/identity.json:21]: This level of detail is too much. Let's refer to the inline documentation instead, like this:

```html
The <code>redirect_url</code> parameter now supports a loopback address, see <a href="https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/identity#getting_the_redirect_url">Getting the redirect URL</a>.
```

**Review by Rob--W — CHANGES_REQUESTED** [review]

File: webextensions/api/identity.json [hunk 2]
```diff
@@ -12,9 +12,15 @@
...
-              "firefox": {
-                "version_added": "53"
-              },
+              "firefox": [
+                {
+                  "version_added": "53"
+                },
+                {
+                  "version_added": "86",
+                  "notes": "The <code>redirect_url</code> parameter now supports a loopback address, see <a href='https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/identity#getting_the_redirect_url'>Getting the redirect URL</a> for details."
```
Comment (Rob--W) [inline webextensions/api/identity.json:21]: The `redirect_url` property belongs to the `launchWebAuthFlow` method, not here.

File: webextensions/api/identity.json [hunk 3]
```diff
@@ -45,6 +51,10 @@
                   "version_added": "53",
                   "version_removed": "63",
```
Comment (Rob--W) [inline webextensions/api/identity.json:52]: This is rendered incorrectly, as "Removed in 63 and later": https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/identity/launchWebAuthFlow#browser_compatibility

Could you change this to `"version_added": 63",` and note that the parameter is optional? It has the same meaning and would be rendered correctly.

File: webextensions/api/identity.json [hunk 4]
```diff
@@ -45,6 +51,10 @@
...
                   "notes": "The <code>redirect_uri</code> URL parameter is required."
+                },
+                {
+                  "version_added": "75",
+                  "notes": "The <code>redirect_uri</code> URL parameter must be set to the URL returned by <code>identity.getRedirectURL()</code>."
```
Comment (Rob--W) [inline webextensions/api/identity.json:57]: With the lot of comments about `redirect_url`, how about adding a new row? `redirect_url`.

Note that Chrome, Opera and Edge does not support the property; their documentation is at https://developer.chrome.com/docs/extensions/reference/identity/#type-WebAuthFlowDetails

**Review by Rob--W — CHANGES_REQUESTED** [review]

**Comment by rebloor** [comment]
@Rob--W Thanks for the feedback, would you like to confirm the changes are expected?

**Review by Rob--W — APPROVED** [review]
Looks great, thanks! In this form, it is very apparent that the `redirect_url` parameter has gone through several revisions.

**Pull request merged**